src/mlpack/tests/kernel_svm_test.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file tests/kernel_svm_test.cpp
+ * @author Himanshu Pathak
+ *
+ * Test the Kernel SVM class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/kernel_svm/kernel_svm.hpp>
+#include <ensmallen.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::svm;
+using namespace mlpack::distribution;
+
+BOOST_AUTO_TEST_SUITE(KernelSVMTest);
+
+BOOST_AUTO_TEST_CASE(KernelSVMGaussianTest)
+{
+  // Generate a two-Gaussian dataset.
+  GaussianDistribution g1(arma::vec("1.0 1.0 1.0"), arma::eye<arma::mat>(3, 3));
+  GaussianDistribution g2(arma::vec("9.0 9.0 9.0"), arma::eye<arma::mat>(3, 3));
+
+  arma::mat data(3, 1000);
+  arma::Row<size_t> responses(1000);
+  for (size_t i = 0; i < 500; ++i)
+  {
+    data.col(i) = g1.Random();
+    responses[i] = -1;
+  }
+  for (size_t i = 500; i < 1000; ++i)
+  {
+    data.col(i) = g2.Random();
+    responses[i] = 1;
+  }
+  const std::string l = "linear";
+  // Now train a logistic regression object on it.
+  KernelSVM<> svm(data, responses, 2, 1.0, 10.0, l,false, 10);
+
+  // Ensure that the error is close to zero.
+  const double acc = svm.ComputeAccuracy(data, responses);
+  BOOST_REQUIRE_CLOSE(acc, 100.0, 0.3); // 0.3% error tolerance.
+
+  // Create a test set.
+  for (size_t i = 0; i < 500; ++i)
+  {
+    data.col(i) = g1.Random();
+    responses[i] = -1;
+  }
+  for (size_t i = 500; i < 1000; ++i)
+  {
+    data.col(i) = g2.Random();
+    responses[i] = 1;
+  }
+
+  // Ensure that the error is close to zero.
+  const double testAcc = svm.ComputeAccuracy(data, responses);
+
+  BOOST_REQUIRE_CLOSE(testAcc, 100.0, 0.6); // 0.6% error tolerance.
```

Comment:
Accuracy 0 seems strange, did you print the actual predictions, are they Inf? Also might be a good strategy to plot the data as well to see if this is the right kernel.